src/data_retrieval/__main__.py

```diff
@@ -57,52 +58,87 @@ def get_cds_era5():
     file_name: str = (
         "era5_total_precipitation_global_1981_2023_all_months.grib"
     )
-    output_path: str = os.path.expanduser("~/Downloads/era5_global_data")
-    setup_output_path(output_path)
+    blob_path = "/raw/glb/era5/era5-total-precipitation-1981-2023.grib"
 
     # Define the years and months for the single request
-    # From 1981 to 2023 try extra 2024 next time
     years: List[int] = list(range(1981, 2024))
-
-    # All months
     months: List[int] = list(range(1, 13))
 
     # Download the data for the entire globe with a single request
-    download_era5_cds(years, months, output_path, file_name)
+    data_stream = download_era5_cds(years, months, file_name)
+
+    # Define the blob path for upload
+
+    # Upload the data to the cloud
+    upload_stream(
+        sas_token, container_name, storage_account, data_stream, blob_path
+    )
+
+
+def upload_to_cloud(
+    data_streams,
+    sas_token,
+    container_name,
+    storage_account,
+    country_name,
+    years,
+):
+    for year, data_stream in zip(years, data_streams):
+
+        filename = (
+            f"{country_name.lower().replace(' ', '_')}_forecast_{year}.grib"
+        )
+        blob_path = f"/raw/{country_name}/mars/{filename}"
+
+        upload_stream(
+            sas_token, container_name, storage_account, data_stream, blob_path
+        )
 
 
 def get_mars(iso: str):
+    logger.info(f"Retrieving data for ISO code: {iso}")
     static_df: pd.DataFrame = get_country_bbox_df()
     country_df: pd.DataFrame = static_df[static_df["iso"].isin([iso.upper()])]
 
     if country_df.empty:
-        logger.error(
-            f"{iso} is not a valid ISO code. See the docs for the list supported counbtries"  # noqa: E501
+        error_msg = (
+            f"{iso} is not a valid ISO code. "
+            "Please refer to the supported countries list."
```

Comment:
I see what you did there, but it is a bit messy. You do not usually split the string like that. It can be confusing for other collaborators. The `  # noqa: E501` disables the long-line lint check for single like, which is a common practise and is easier to read.